chain/owasm/resolver.go

```diff
@@ -123,25 +130,40 @@ func (r *resolver) resolveRequestExternalData(vm *exec.VirtualMachine) int64 {
 	return 0
 }
 
+func (r *resolver) getExternalDataFromCache(externalDataID int64, validatorIndex int64) ([]byte, error) {
+	if r.cachedata.externalDataID == externalDataID && r.cachedata.validatorIndex == validatorIndex {
+		return r.cachedata.data, nil
+	}
+	externalData, err := r.env.GetExternalData(externalDataID, validatorIndex)
+	r.cachedata = cache{
+		externalDataID: externalDataID,
+		validatorIndex: validatorIndex,
+		data:           externalData,
+	}
+
+	return externalData, err
+}
+
 func (r *resolver) resolveGetExternalDataSize(vm *exec.VirtualMachine) int64 {
 	externalDataID := GetLocalInt64(vm, 0)
 	validatorIndex := GetLocalInt64(vm, 1)
-	// TODO: ExternalData should be cached for both this function and the one below.
-	externalData, err := r.env.GetExternalData(externalDataID, validatorIndex)
+
+	externalData, err := r.getExternalDataFromCache(externalDataID, validatorIndex)
```

Comment:
If user call for `externalDataID = 0` and `validatorIndex = 0`, won't we incorrectly think it's already in the cache?